benchmark/bench_matrix_ops.jl

```diff
@@ -0,0 +1,45 @@
+module BenchMatrixOps
+
+import Random
+using BenchmarkTools
+using LinearAlgebra
+using StaticArrays
+
+const suite = BenchmarkGroup()
+const matrix_sizes = if haskey(ENV, "GITHUB_EVENT_PATH")
+    (1, 2, 3, 4, 10, 20)
```

Comment:
* `matrix_sizes = 1:20` => total CI time = 37 min https://github.com/JuliaArrays/StaticArrays.jl/runs/456805444
* `matrix_sizes = (1, 2, 3, 4, 10, 20)` => total CI time = 15 min https://github.com/JuliaArrays/StaticArrays.jl/runs/456815324

Reply:
Great :+1:

If we're already at 15 min, I worry a bit about will happen for a comprehensive benchmark suite :grimacing:

Reply:
(I guess using the same treatment for QR would probably help with that 15 mins though)

Reply:
Yeah, I have the same worry. I guess we can't do the "full sweep" in CI and be a bit less greedy about the parameters we try.

An option to reduce CI time is to tune benchmarks separately and store the JSON file somewhere. But I think it's easy to get it stale and cumbersome to add benchmarks...

BTW, a less ad-hoc way to do this kind of filtering is to use `@tagged` from BenchmarkToos.jl: https://github.com/JuliaCI/BenchmarkTools.jl/blob/master/doc/manual.md#indexing-into-a-benchmarkgroup-using-tagged I don't think PkgBenchmark.jl support this though.

Reply:
I also wonder how much of this is compile time; some of the algorithms are probably quite terrible for N=20.

Reply:
Yeah, one thing I can't copy easily from `perf/*.jl` is the compile-time benchmarks. It'd be great if BenchmarkTools reports compile-time while doing the tuning. I don't think it is difficult but just tedious...